torch/_higher_order_ops/auto_functionalize.py

```diff
@@ -365,7 +403,19 @@ def __call__(
 auto_functionalized_v2.fallthrough(DispatchKey.AutogradCUDA)
 
 
-def can_auto_functionalize(op: OperatorBase) -> bool:
+def can_auto_functionalize(
+    op: Union[OperatorBase, HopInstance],
+) -> bool:
+    if isinstance(op, HopInstance):
+        # HOPs that implement gen_schema and schema is not functional are auto_functionalizable.
+        def _has_gen_schema(op: HigherOrderOperator):
+            method = "gen_schema"
+            return hasattr(type(op), method) and getattr(
+                type(op), method
+            ) is not getattr(HigherOrderOperator, method)
+
+        return _has_gen_schema(op._op) and op._schema.is_mutable
```

Comment:
nit: You also probably want the other checks that we do on schema below (lines 431-450)?